loadtest/control/gencontroller/actions.go

```diff
@@ -71,6 +72,51 @@ func (c *GenController) createTeam(u user.User) (res control.UserActionResponse)
 	return control.UserActionResponse{Info: fmt.Sprintf("created team %s", id)}
 }
 
+func (c *GenController) createCPAField(u user.User) (res control.UserActionResponse) {
+	if !st.inc(StateTargetCPAFields, c.config.NumCPAFields) {
+		return control.UserActionResponse{Info: "target number of custom profile fields reached"}
+	}
+	defer func() {
+		if res.Err != nil || res.Warn != "" {
+			st.dec(StateTargetCPAFields)
+		}
+	}()
+
+	cpaField := &model.PropertyField{
+		Name: control.PickRandomWord() + "_" + control.PickRandomWord(),
+		Type: model.PropertyFieldTypeText,
+	}
+	// Only sysadmin can create CPA fields
+	field, err := c.sysadmin.CreateCPAField(cpaField)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
```

Comment:
I would rather skip this action for non-admins here, I think it would be semantically more correct. So something like:
```suggestion
	// Only sysadmin can create CPA fields
	isSysAdmin, err := u.IsSysAdmin()
	if err != nil {
		return UserActionResponse{Err: NewUserError(err)}
	}
	
	if !isSysAdmin {
		return UserActionResponse{Warn: "not an admin user, unable to create a CPA field"}
	}

	field, err := c.sysadmin.CreateCPAField(cpaField)
	if err != nil {
		return control.UserActionResponse{Err: control.NewUserError(err)}
	}
```

Reply:
I had written basically the same code you have above, based on your previous comments. Then as I continued investigating, it doesn't appear that user `u` is ever a sysadmin and a sysadmin user is passed separately. Which is why I used `c.sysadmin.CreateCPAField`.

Looking at this code
https://github.com/mattermost/mattermost-load-test-ng/blob/master/api/agent.go#L446-L455

Let me know, I'm probably misunderstanding something. Hard when you don't get to run it... :)

Reply:
I believe this relies on the `UsersConfiguration.PercentOfUsersAreAdmin` setting. See https://github.com/mattermost/mattermost-load-test-ng/blob/1e61b93969d2a2c7705981c3d62b6d0c30899ecf/api/agent.go#L373-L376 and https://github.com/mattermost/mattermost-load-test-ng/blob/1e61b93969d2a2c7705981c3d62b6d0c30899ecf/api/agent.go#L392-L395 But I'm not 100% sure, so let me double-check this first.

Reply:
Ok...you sold me. Adding the sysadmin check. I missed that.